webapp/components/test/test-runs-query-builder.html

```diff
@@ -22,29 +22,33 @@
 import '../test-runs-query-builder.js';
 import { PolymerElement } from '../../node_modules/@polymer/polymer/polymer-element.js';
 import { Channels } from '../product-info.js';
+
 suite('TestRunsQueryBuilder', () => {
   let queryBuilder, sandbox;
 
-  setup(() => {
+  suiteSetup(() => {
     sandbox = sinon.sandbox.create();
     // Spoof an empty result for /api/shas to speed up tests.
-    const realFetch = window.fetch;
+    // This is done in suiteSetup/suiteTeardown to avoid async fetches slipping
+    // through between tests.
+    const ignore = new RegExp('/api/(shas|versions)');
     sandbox.stub(window, 'fetch', url => {
-      const ignore = new RegExp('/api/(shas|versions)');
       if (ignore.test(url.pathname)) {
         return Promise.resolve(new Response('[]'));
       }
-      return realFetch(url);
+      throw url;
```

Comment:
The `fetch` method which this stubs out never throws an exception, it rejects. Fix would be similar to https://github.com/web-platform-tests/wpt.fyi/pull/1354.

Reply:
This wasn't intended to match the behaviour of `fetch`. Rather, an unusual exception is raised to show that there is a programming error. If we were to test the error handling of the component, we would then set up another set of expectations and reject the promise if the URL matches one of the failure expectations.

Reply:
In other words, this error shouldn't be reached while executing the tests, that would be an error in the test code? Throwing an error with a more elaborate error message might make that clearer.

Reply:
Yes. And you're right; I should write a message instead of just throwing the url.